src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -228,6 +228,7 @@ CreateModifyPlan(Query *originalQuery, Query *query,
 	distributedPlan->masterQuery = NULL;
 	distributedPlan->routerExecutable = true;
 	distributedPlan->hasReturning = false;
+	distributedPlan->targetRelationId = ExtractFirstDistributedTableId(query);
```

Comment:
This is probably not an issue, but if we do modifying CTEs or modifies with subqueries with router executor (or something fancier), would that always give the correct target relationId?

I wonder whether we should write a new function which gets the  `	rte = rt_fetch(query->resultRelation, query->rtable);  	Assert(rte->rtekind == RTE_RELATION); return rte->relid`  or such